pkg/protocol/engine/mempool/v1/mempool.go

```diff
@@ -197,6 +197,42 @@ func (m *MemPool[VoteRank]) stateDiff(slot iotago.SlotIndex) (*StateDiff, error)
 	return lo.Return1(m.stateDiffs.GetOrCreate(slot, func() *StateDiff { return NewStateDiff(slot, kv) })), nil
 }
 
+// Reset resets the component to a clean state as if it was created at the last commitment.
+func (m *MemPool[VoteRank]) Reset() {
+	stateDiffsToDelete := make([]iotago.SlotIndex, 0)
+	m.stateDiffs.ForEachKey(func(slot iotago.SlotIndex) bool {
+		if slot > m.lastEvictedSlot {
+			stateDiffsToDelete = append(stateDiffsToDelete, slot)
+		}
+
+		return true
+	})
+
+	attachmentsToDelete := make([]iotago.SlotIndex, 0)
+	m.attachments.ForEach(func(slot iotago.SlotIndex, _ *shrinkingmap.ShrinkingMap[iotago.BlockID, *SignedTransactionMetadata]) {
+		if slot > m.lastEvictedSlot {
+			attachmentsToDelete = append(attachmentsToDelete, slot)
+		}
+	})
+
+	for _, slot := range stateDiffsToDelete {
+		if stateDiff, deleted := m.stateDiffs.DeleteAndReturn(slot); deleted {
+			if err := stateDiff.Reset(); err != nil {
```

Comment:
Why do we need to reset the state diff instance as well? I thought we remove the instance completely anyway.

Reply:
The StateDiffs roll something back on disk